docs/source/tracking/data-api.rst

```diff
@@ -0,0 +1,227 @@
+MLflow Data Module Tutorial
+===========================
+
+The ``mlflow.data`` module is an integral part of the MLflow ecosystem, designed to enhance your machine learning workflow.
+This module enables you to record and retrieve dataset information during model training and evaluation, leveraging MLflow's tracking capabilities.
+
+Key Interfaces
+--------------
+
+There are two main abstract components associated with the ``mlflow.data`` module:
+
+Dataset 
+^^^^^^^
+
+The ``Dataset`` abstraction is a metadata tracking object that holds the information about a given logged dataset.
+
+The information stored within a ``Dataset`` object includes features, targets, and predictions, along with 
+metadata like the dataset's name, digest (hash), schema, and profile. You can log this metadata using the :py:func:`mlflow.log_input` API. 
+The module provides functions to construct :py:class:`mlflow.data.dataset.Dataset` objects from various data types.
+
+There are a number of concrete implementations of this abstract class, including:
+
+- :py:class:`mlflow.data.spark_dataset.SparkDataset`
+- :py:class:`mlflow.data.pandas_dataset.PandasDataset`
+- :py:class:`mlflow.data.numpy_dataset.NumpyDataset`
+- :py:class:`mlflow.data.huggingface_dataset.HuggingFaceDataset`
+- :py:class:`mlflow.data.spark_dataset.SparkDataset`
+- :py:class:`mlflow.data.tensorflow_dataset.TensorFlowDataset`
+
+The following example demonstrates how to construct a :py:class:`mlflow.data.pandas_dataset.PandasDataset` object from a Pandas DataFrame:
+
+.. code-block:: python
+
+    import mlflow.data
+    import pandas as pd
+    from mlflow.data.pandas_dataset import PandasDataset
+
+
+    CSV_DELIMITER = ";"
+
+    dataset_source_url = "https://raw.githubusercontent.com/mlflow/mlflow/master/tests/datasets/winequality-white.csv"
+    raw_data = pd.read_csv(dataset_source_url, delimiter=CSV_DELIMITER)
+
+    # Create an instance of a PandasDataset
+    dataset = mlflow.data.from_pandas(
+        raw_data, source=dataset_source_url, name="wine quality - white", targets="quality"
+    )
+
+DatasetSource
+^^^^^^^^^^^^^
+
+The ``DatasetSource`` is a component of a given Dataset object, providing a linked lineage to the original source of the data.
+
+The ``DatasetSource`` component of a ``Dataset`` represents the source of a dataset, such as a directory in S3, a Delta Table, or a URL. 
+It is referenced in the ``Dataset`` for understanding the origin of the data. The ``DatasetSource`` of a logged 
+dataset can be retrieved using the ``mlflow.data.get_source()`` API.
+
+Example Usage
```

Comment:
Can we also mention that autolog logs dataset automatically for some flavors? I think it's more common to log it as a part of training/evaluation rather than creating a dedicated run for logging dataset.

Reply:
Added a tip and a note